### PR TITLE
Keep router failure provenance untrusted

### PR DESCRIPTION
**Pull request by basnijholt**
## Summary
- avoid stamping router fallback/failure notices with original-sender provenance
- ignore stale original-sender metadata on router-authored failure chatter unless the router message actually targets a responder
- document the plain hook vs hook_dispatch mention-gate distinction that came up in #1078 review

## Review comment triage from #1078
- Sourcery managed-requester test gap was already fixed before #1078 merged.
- Gemini/Greptile hook-bypass suggestions were intentionally not applied because plain hook sends follow normal routing, while hook_dispatch bypasses through ingress_policy. Added a code comment here.
- Gemini managed-requester provenance suggestion would make agent relays look user-authored, so this keeps that guard and hardens the failure-notice edge case instead.

## Tests
- uv run --no-sync pytest tests/test_hook_sender.py::test_prepare_dispatch_uses_trusted_router_context_for_router_relays tests/test_multi_agent_bot.py::TestAgentBot::test_trusted_internal_router_relays_use_gate_bypass tests/test_voice_command_processing.py::test_router_processes_own_voice_transcriptions tests/test_voice_command_processing.py::test_router_processes_own_sidecar_commands_using_original_sender tests/test_bot_scheduling.py::TestCommandHandling::test_router_error_prevents_team_formation tests/test_routing_regression.py::TestRoutingRegression::test_router_relay_failure_does_not_stamp_original_sender -n 0 --no-cov -v
- uv run --no-sync pytest tests/test_bot_scheduling.py tests/test_routing_regression.py -n 0 --no-cov -v
- uv run --no-sync pre-commit run --files src/mindroom/turn_controller.py tests/test_bot_scheduling.py tests/test_routing_regression.py
- uv run --no-sync pytest -n 0 --no-cov -q

## Summary by Sourcery

Harden router relay provenance handling so failure and fallback messages do not inherit untrusted original-sender metadata, while preserving trusted router handoffs for specific routed contexts.

Bug Fixes:
- Ensure router-authored relay failures and error notices do not stamp or trust ORIGINAL_SENDER metadata unless the event represents a trusted relay to another responder.
- Prevent router relay failure messages from being treated as trusted internal handoffs that could bypass mention-based routing gates.
- Ensure router error messages cannot trigger team formation even when ORIGINAL_SENDER metadata is present.

Enhancements:
- Introduce centralized logic to decide when original-sender metadata should be treated as trusted based on sender identity, content shape, and source kind.
- Clarify the behavior difference between plain hook routing and hook_dispatch-based gate bypass via an inline code comment.

Tests:
- Add regression tests covering router relay failure provenance, router error handling with and without ORIGINAL_SENDER metadata, and trusted internal relay gating behavior.

### CODE REVIEW

File: src/mindroom/turn_controller.py [hunk 1]
```diff
@@ -253,6 +266,53 @@ def _managed_entity_name_for_sender(self, sender_id: str, *, include_router: boo
...
+                return True
+
+        body = content.get("body")
+        if not isinstance(body, str):
+            return False
+        if body.lstrip().startswith("@"):
+            return True
+        return any(
+            name != ROUTER_AGENT_NAME and matrix_id.full_id in body for name, matrix_id in registry.current_ids.items()
+        )
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply src/mindroom/turn_controller.py:294]: Fixed in e6eb1ce76: router-authored provenance is now trusted only when the body or m.mentions resolves to a configured non-router entity, and the regression covers plain-user @mentions staying untrusted.

File: tests/test_routing_regression.py [hunk 2]
```diff
@@ -630,6 +630,75 @@ async def test_router_relay_does_not_stamp_managed_requester_as_original_sender(
...
+        message_event = MagicMock(spec=nio.RoomMessageText)
+        message_event.sender = "@user:localhost"
+        message_event.body = "Please route this"
+        message_event.event_id = "$router_failure_message"
+        message_event.server_timestamp = 1000
+        message_event.source = {"content": {"body": "Please route this"}}
+
+        await router_bot._turn_controller._execute_router_relay(
+            mock_room,
+            message_event,
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply tests/test_routing_regression.py:696]: Fixed in e6eb1ce76: the router failure regression now seeds ORIGINAL_SENDER_KEY on the incoming event and still asserts the failure response drops it.

**Comment by basnijholt** [comment]
Checked Sourcery’s high-level note about `_should_trust_original_sender_metadata` returning `True` for non-router managed senders. I left that path unchanged intentionally: it is only reachable for exact current managed Matrix IDs, and those non-router agents are the trusted relay boundary for scheduled/hook/media payload metadata. User-authored spoofed `ORIGINAL_SENDER_KEY` content is rejected before this path because the sender is not a managed entity; router-authored plain message relays are the special case now constrained by configured mention resolution.
